eth/tracers/api_test.go

```diff
@@ -421,6 +431,18 @@ func TestTraceTransaction(t *testing.T) {
 	}) {
 		t.Error("Transaction tracing result is different")
 	}
+
+	// test TraceTransaction for a historical transaction
+	result2, err := api.TraceTransaction(context.Background(), common.HexToHash("0xACDC"), nil)
```

Comment:
What exactly is the mock testing? It seems like we're just checking that the testing code returns what we put in the testing code.

Reply:
The mock isn't really testing anything, but it enables us to verify that the Sequencer's `TraceTransaction` will fallback to the historical rpc if the tx is not found.

Reply:
^ which I think is the main thing to test. The same approach was used to test `TraceBlock` [below](https://github.com/ethereum-optimism/op-geth/pull/20/files#diff-a63a59094cfdc02a16570494304f3253c68fc35452b395f22d7652b9c83fdb0aR490-R494).
In order to go beyond this, I think we'd need to initialize another node and chain, which will add a lot more plumbing code to test what's already being tested in the rest of this file.